app/src/main/java/org/oppia/app/splash/SplashActivityPresenter.kt

```diff
@@ -7,20 +7,24 @@ import androidx.lifecycle.Observer
 import androidx.lifecycle.Transformations
 import org.oppia.app.R
 import org.oppia.app.activity.ActivityScope
-import org.oppia.app.model.OnboardingFlow
+import org.oppia.app.deprecation.AutomaticAppDeprecationNoticeDialogFragment
+import org.oppia.app.model.AppStartupState
+import org.oppia.app.model.AppStartupState.StartupMode
 import org.oppia.app.onboarding.OnboardingActivity
 import org.oppia.app.profile.ProfileActivity
-import org.oppia.domain.onboarding.OnboardingFlowController
+import org.oppia.domain.onboarding.AppStartupStateController
 import org.oppia.util.data.AsyncResult
 import org.oppia.util.logging.Logger
 import javax.inject.Inject
 
+private const val AUTO_DEPRECATION_NOTICE_DIALOG_FRAGMENT_TAG = "auto_deprecation_notice_dialog"
+
 /** The presenter for [SplashActivity]. */
 @ActivityScope
 class SplashActivityPresenter @Inject constructor(
```

Comment:
Tests to add:
- Open app, initial open, expiration enabled, before expiration date, intents to onboarding flow
- Open app, initial open, expiration enabled, after expiration date, intents to deprecation dialog
- Open app, initial open, expiration enabled, after expiration date, click on close dialog, ends splash activity
- Open app, initial open, expiration disabled, after expiration date, intents to onboarding flow
- Open app, reopen, expiration enabled, before expiration date, intents to profile chooser
- Open app, reopen, expiration enabled, after expiration date, intents to deprecation dialog